presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -381,6 +383,23 @@ public void addInput(Page page)
     }
 
     @Override
+    public ListenableFuture<?> startMemoryRevoke()
+    {
+        if (aggregationBuilder != null) {
```

Comment:
It's not required in this operator. You could call `startMemoryRevoke` and `finishMemoryRevoke` arbitrary number of times in whatever order you want, so it's a bad place for such a `checkState`